### PR TITLE
update transform history to store coordinated places as array list

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

File: src/main/java/emissary/core/TransformHistory.java [hunk 1]
```diff
@@ -118,13 +135,11 @@ public List<History> getHistory() {
      * @return last place visited
      */
     @Nullable
-    public String lastVisit() {
-        List<String> historyList = get();
-        final int sz = historyList.size();
-        if (sz == 0) {
+    public History lastVisit() {
```
Comment (jpdahlke) [inline src/main/java/emissary/core/TransformHistory.java:138]: They key piece here is that we no longer use the `get()` method when determining the last place we visited. That method should be reserved for use when formatting a final transform history as requested by the BaseDataObject/DropOff.

File: src/test/java/emissary/core/TransformHistoryTest.java [hunk 2]
```diff
@@ -151,7 +151,7 @@ void testToString() {
 
         String newline = System.getProperty("line.separator");
         StringBuilder expected = new StringBuilder();
-        expected.append("transform history (6) :").append(newline);
+        expected.append("transform history (5) :").append(newline);
```
Comment (jpdahlke) [inline src/test/java/emissary/core/TransformHistoryTest.java:154]: This side-effect is interesting. Do we prefer a total count to include coordinated places?

Comment (fbruton) [reply src/test/java/emissary/core/TransformHistoryTest.java:154]: I prefer that coordinated places be included.

**Review by jpdahlke — APPROVED** [review]

**Review by fbruton — APPROVED** [review]

**Comment by fbruton** [comment]
Configured a coordination place that includes `DelayPlace` and realized that the coordinated place was not indented.
```
   currentForms: []
   filetype: XML
   transform history (5) :
     UNKNOWN.FILE_PICK_UP.INPUT.http://localhost:8001/FilePickUpPlace$5050
     UNKNOWN.UNIXFILE.ID.http://localhost:8001/UnixFilePlace$2050
     XML.TESTCOORDINATE.COORDINATE.http://localhost:8001/CoordinationPlace$3050
     TESTCOORDINATE.DELAY.ANALYZE.http://localhost:8001/DelayPlace$5050
     TESTCOORDINATE.DROP_OFF.IO.http://localhost:8001/DropOffPlace$9990
```
CoordinationPlace.cfg
```
# Test to set up a coordinate place that needs an upper and a lower
URL = "@{emissary.node.scheme}://@{emissary.node.name}:@{emissary.node.port}"
SERVICE_KEY = "XML.TESTCOORDINATE.COORDINATE.@{URL}/CoordinationPlace$3050"

SERVICE_COORDINATION = "DelayPlace"

OUTPUT_FORM = "TESTCOORDINATE"
```

**Comment by fbruton** [comment]
Expected the coordination place to run in the order listed in `Coordination.cfg`. Current behavior appears to be routing based on the output form.
```
   creationTimestamp: 2024-08-01T11:55:29.266168Z
   currentForms: []
   filetype: XML
   transform history (13) :
     UNKNOWN.FILE_PICK_UP.INPUT.http://localhost:8001/FilePickUpPlace$5050
     UNKNOWN.UNIXFILE.ID.http://localhost:8001/UnixFilePlace$2050
     XML.TESTCOORDINATE.COORDINATE.http://localhost:8001/CoordinationPlace$3050
     UPPER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToLowerPlace$6010
     LOWER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToUpperPlace$5010
     UPPER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToLowerPlace$6010
     LOWER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToUpperPlace$5010
     UPPER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToLowerPlace$6010
     LOWER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToUpperPlace$5010
     UPPER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToLowerPlace$6010
     FINI.DELAY.ANALYZE.http://localhost:8001/DelayPlace$5050
     FINI.DROP_OFF.IO.http://localhost:8001/DropOffPlace$9990
     [TESTCOORDINATE].DROP_OFF.IO.http://localhost:8001/DropOffPlace
```

**Comment by dev-mlb** [comment]
@fbruton try adding `UPDATE_TRANSFORM_HISTORY = true` to your config and see if that helps

**Comment by fbruton** [comment]
> UPDATE_TRANSFORM_HISTORY = true

```
   creationTimestamp: 2024-08-01T12:52:34.814062Z
   currentForms: []
   filetype: XML
   transform history (8) :
     UNKNOWN.FILE_PICK_UP.INPUT.http://localhost:8001/FilePickUpPlace$5050
     UNKNOWN.UNIXFILE.ID.http://localhost:8001/UnixFilePlace$2050
     XML.TESTCOORDINATE.COORDINATE.http://localhost:8001/CoordinationPlace$3050
      XML.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToLowerPlace
      LOWER_CASE.SERVICE_IS_COORDINATED.TRANSFORM.http://localhost:8001/ToUpperPlace
      UPPER_CASE.DELAY.ANALYZE.http://localhost:8001/DelayPlace
     TESTCOORDINATE.DELAY.ANALYZE.http://localhost:8001/DelayPlace$5050
     TESTCOORDINATE.DROP_OFF.IO.http://localhost:8001/DropOffPlace$9990
```
Coorindation.cfg
```
# Test to set up a coordinate place that needs an upper and a lower
URL = "@{emissary.node.scheme}://@{emissary.node.name}:@{emissary.node.port}"
SERVICE_KEY = "XML.TESTCOORDINATE.COORDINATE.@{URL}/CoordinationPlace$3050"

SERVICE_COORDINATION = "ToLowerPlace"
SERVICE_COORDINATION = "ToUpperPlace"
SERVICE_COORDINATION = "DelayPlace"


OUTPUT_FORM = "TESTCOORDINATE"

UPDATE_TRANSFORM_HISTORY = "true"
PUSH_OUTPUT_FORM = "false"
```

**Comment by fbruton** [comment]
I think the change is good now. Updated the SERVICE_PROXY config entries for ToUpper and ToLower to `SERVICE_IS_COORDINATED` and getting the expected transform history.

```
   creationTimestamp: 2024-08-01T14:44:44.999481Z
   currentForms: []
   filetype: XML
   transform history (6) :
     UNKNOWN.FILE_PICK_UP.INPUT.http://localhost:8001/FilePickUpPlace$5050
     UNKNOWN.UNIXFILE.ID.http://localhost:8001/UnixFilePlace$2050
     XML.TESTCOORDINATE.COORDINATE.http://localhost:8001/CoordinationPlace$3050
     TESTCOORDINATE.DELAY.ANALYZE.http://localhost:8001/DelayPlace$5050
     TESTCOORDINATE.DROP_OFF.IO.http://localhost:8001/DropOffPlace$9990
     [UPPER_CASE].DROP_OFF.IO.http://localhost:8001/DropOffPlace
```
CoordinationPlace.cfg
```
# Test to set up a coordinate place that needs an upper and a lower
URL = "@{emissary.node.scheme}://@{emissary.node.name}:@{emissary.node.port}"
SERVICE_KEY = "XML.TESTCOORDINATE.COORDINATE.@{URL}/CoordinationPlace$3050"

SERVICE_COORDINATION = "ToLowerPlace"
SERVICE_COORDINATION = "ToUpperPlace"
SERVICE_COORDINATION = "DelayPlace"


OUTPUT_FORM = "TESTCOORDINATE"

#UPDATE_TRANSFORM_HISTORY = "true"
#PUSH_OUTPUT_FORM = "false"
```

**Review by fbruton — APPROVED** [review]

**Pull request merged**